pkg/rbac/parser.go

```diff
@@ -44,8 +47,54 @@ type Rule struct {
 	URLs      []string `marker:"urls,optional"`
 }
 
+// ruleKey represents the resources and non-resources a Rule applies.
+type ruleKey string
```

Comment:
this can just be a struct:

```go
// ruleKey allows deduping rbac rules by groups/resources/urls combinations
// the keys are strings joined by `,`s to allow for use as a map key
type ruleKey struct {
  Groups string
  Resources string
  URLs string
}
```

Reply:
I still don't see why using a struct here is better than a string.
We just need something to identify a rule.

Reply:
it's a bit clearer as to how it's composed, and a bit harder to mess up + one fewer join operation.

Reply:
Fixed it! Thanks!